src/objects.js

```diff
@@ -1,42 +1,54 @@
 const createPerson = (name, age) => {
-  // your code here
+  return {
+    name: name,
+    age: age
+  };
 };
 
 const getName = object => {
-  // your code here
+  return object.name;
 };
 
 const getProperty = (property, object) => {
-  // your code here
+  return object[property];
 };
 
-const hasProperty = (property, object) => {
-  // your code here
-};
+function hasProperty(property, object) {
+  return object. hasOwnProperty (property);
+}
 
 const isOver65 = person => {
-  // your code here
+  return person.age > 65;
 };
 
 const getAges = people => {
-  // your code here
+  let result = people.map(a => a.age);
+  return result;
+  //worked but don't understand how the function inside .map's brackets really works
 };
 
 const findByName = (name, people) => {
-  // your code here
+  return people.find(x => x.name === name);
 };
 
 const findHondas = cars => {
-  // your code here
+  return cars.filter(x => x.manufacturer === 'Honda');
 };
 
 const averageAge = people => {
-  // your code here
+  const listOfAges = people.map(a => a.age);
+  const result = listOfAges.reduce((total, currentAmount)=>total+currentAmount);
+  return result / listOfAges.length;
 };
 
 const createTalkingPerson = (name, age) => {
-  // your code here
+  return {
+    name: name,
+    age: age,
```

Comment:
Theres a bit of shorthand you can do here, not essential but good to know about:
```suggestion
    name,
    age,
```